lib/types.ts

```diff
@@ -330,13 +330,13 @@ type DeepModify<T> =
         ? true | DeepModify<T[K]>
         : true;
     }
-  | Array<DeepModify<T>>
-  | Promise<DeepModify<T>>
-  | Set<DeepModify<T>>
-  | ReadonlySet<DeepModify<T>>
-  | WeakSet<DeepModify<T>>
-  | Map<any, DeepModify<T>>
-  | WeakMap<any, DeepModify<T>>;
+  | (T extends Array<infer E> ? Array<DeepModify<E>> : never)
+  | (T extends Promise<infer E> ? Promise<DeepModify<E>> : never)
+  | (T extends Set<infer E> ? Set<DeepModify<E>> : never)
+  | (T extends ReadonlySet<infer E> ? ReadonlySet<DeepModify<E>> : never)
+  | (T extends WeakSet<infer E> ? WeakSet<DeepModify<E>> : never)
+  | (T extends Map<any, infer E> ? Map<any, DeepModify<E>> : never)
+  | (T extends WeakMap<any, infer E> ? WeakMap<any, DeepModify<E>> : never);
```

Comment:
Should these last 2 be

```typescript
  | (T extends Map<infer K, infer E> ? Map<K, DeepModify<E>> : never)
  | (T extends WeakMap<infer K, infer E> ? WeakMap<K, DeepModify<E>> : never);
```

I kinda like this union type style vs the deep nested conditional style of DeepPick/DeepOmit.


Does this change belong here at all? Are there any tests for this change?

Reply:
That's a good catch! 👀

I think it doesn't really change anything because we are focusing more on values rather than keys

Anyway I need to add tests for at least all maps

Reply:
However something is covered here – https://github.com/krzkaczor/ts-essentials/pull/273/files#diff-8c10bb7fdebd2b6fa3438e87e51411024b7af2ccb5abd52dd83e52cadae49a45R618

See that `number` cannot be used for `Maps` because it was declared `string`

Reply:
I covered it ✅

Now it should be strictly equal to the value